### PR TITLE
IMB-46 Declaration pages

### PR DESCRIPTION
**Pull request by muthrajan**
## What
- Declaration pages - [IMB-46](https://collaboration.homeoffice.gov.uk/jira/browse/IMB-46)
## Why
- part of IMA MVP
## How
- Declaration pages for all 3 different routes and finally submit

## Testing
- tests passing locally

### CODE REVIEW

File: apps/ima/index.js [hunk 1]
```diff
@@ -33,13 +38,68 @@ module.exports = {
...
+    '/declaration-immigration-adviser': {
+      behaviours: [Summary, Submit],
+      sections: require('./sections/summary-data-sections'),
+      fields: [
+        'declaration-immigration-adviser'
+      ],
+      locals: { showSaveAndExit: true },
+      next: '/confirmation'
+    },
+    '/declaration-someone-else': {
```
Comment (mislam987) [inline apps/ima/index.js:96]: Path names do not match those in the latest Figma designs.

File: apps/ima/fields/index.js [hunk 2]
```diff
@@ -2,9 +2,37 @@
...
+    isPageHeading: false,
+    mixin: 'radio-group',
+    labelClassName: 'visuallyhidden',
+    options: [{
+      value: 'yes'
+    }, {
+      value: 'no'
+    }],
     validate: 'required'
   }
```
Comment (mislam987) [inline apps/ima/fields/index.js:37]: Validation messages not showing on top of screen or above mandatory fields if left blank. Same goes for the other mandatory fields.

File: apps/ima/fields/index.js [hunk 3]
```diff
@@ -2,9 +2,37 @@
...
+    isPageHeading: false,
+    mixin: 'radio-group',
+    labelClassName: 'visuallyhidden',
+    options: [{
+      value: 'yes'
+    }, {
+      value: 'no'
+    }],
+    validate: 'required'
+  },
```
Comment (mislam987) [inline apps/ima/fields/index.js:26]: If this checkbox is left as blank, user can progress with the workflow. Is that correct? Same applies for the other declaration checkboxes.

File: apps/ima/views/declaration-immigration-adviser.html [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+ {{#markdown}}declaration-immigration-adviser{{/markdown}}
+ <br><br>
+ <div class="govuk-checkboxes__item">
+    <input class="govuk-checkboxes__input" id="declaration-immigration-adviser" name="declarationImmigrationAdviser" type="checkbox">
+    <label class="govuk-label govuk-checkboxes__label" for="declaration-immigration-adviser">
+      {{#t}}fields.declaration-immigration-adviser.label{{/t}}
+    </label>
+  </div>
+  <br><br>
+  <p>You will not be able to change your answer after you submit.</p>
```
Comment (mislam987) [inline apps/ima/views/declaration-immigration-adviser.html:18]: Should be 'You will not be able to change your answers after you submit.'.

File: apps/ima/views/declaration-person-named.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ {{#markdown}}declaration-person-named{{/markdown}}
+ <br><br>
+ <div class="govuk-checkboxes__item">
+    <input class="govuk-checkboxes__input" id="declaration-person-named" name="declarationPersonNamed" type="checkbox">
+    <label class="govuk-label govuk-checkboxes__label" for="declaration-person-named">
+      {{#t}}fields.declaration-person-named.label{{/t}}
+    </label>
+  </div>
+  <br><br>
+  <p>You will not be able to change your answer after you submit.</p>
```
Comment (mislam987) [inline apps/ima/views/declaration-person-named.html:11]: Should be 'You will not be able to change your answers after you submit'.

File: apps/ima/views/declaration-someone-else.html [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+ {{#markdown}}declaration-someone-else{{/markdown}}
+ <br><br>
+ <div class="govuk-checkboxes__item">
+    <input class="govuk-checkboxes__input" id="declaration-someone-else" name="declarationSomeoneElse" type="checkbox">
+    <label class="govuk-label govuk-checkboxes__label" for="declaration-isomeone-else">
+      {{#t}}fields.declaration-someone-else.label{{/t}}
+    </label>
+  </div>
+  <br><br>
+  <p>You will not be able to change your answer after you submit.</p>
```
Comment (mislam987) [inline apps/ima/views/declaration-someone-else.html:18]: Should be 'You will not be able to change your answers after you submit'.

File: apps/ima/index.js (hunk 1 above)
Comment (muthrajan) [reply apps/ima/index.js]: Path names changed as per Figma designs.

File: apps/ima/fields/index.js (hunk 2 above)
Comment (muthrajan) [reply apps/ima/fields/index.js]: Validation added with messages for all mandatory fields.

File: apps/ima/fields/index.js (hunk 3 above)
Comment (muthrajan) [reply apps/ima/fields/index.js]: Validation with messages created for checkbox

File: apps/ima/views/declaration-immigration-adviser.html (hunk 4 above)
Comment (muthrajan) [reply apps/ima/views/declaration-immigration-adviser.html]: Changed to answers

File: apps/ima/views/declaration-person-named.html (hunk 5 above)
Comment (muthrajan) [reply apps/ima/views/declaration-person-named.html]: Changed to answers

File: apps/ima/views/declaration-someone-else.html (hunk 6 above)
Comment (muthrajan) [reply apps/ima/views/declaration-someone-else.html]: Changed to answers

**Pull request closed without merging**